packages/repository/test/unit/repository-mixin/repository-mixin.test.ts

```diff
@@ -128,3 +128,48 @@ describe('RepositoryMixin', () => {
     expect(componentInstance).to.be.instanceOf(component);
   }
 });
+
+describe('RepositoryMixin dataSource', () => {
+  it('is mixed into the target class', () => {
+    const myApp = new AppWithRepoMixin();
+    expect(typeof myApp.dataSource).to.be.eql('function');
+  });
+
+  it('is not conflict with previous binding key', () => {
+    const myApp = new AppWithRepoMixin();
+    const withoutDataSource = myApp.find('datasources.*');
+    expect(withoutDataSource).to.be.empty();
+  });
+
+  it('binds dataSource to a binding key using the dataSource name property', () => {
+    const myApp = new AppWithRepoMixin();
+    const fooDataSource: juggler.DataSource = new DataSourceConstructor({
+      name: 'foo',
+      connector: 'memory',
+    });
+    myApp.dataSource(fooDataSource);
+    expectDataSourceToBeBound(myApp, fooDataSource, 'foo');
+  });
+
+  it('binds dataSource to a binding key using the name param', () => {
```

Comment:
nitpick: I think the test name would be phrased with `... using the given argument/name/string` instead of `name param`